### ISSUE TITLE
Confusing "channel graph has insufficient capacity" error

### ISSUE THREAD

**Issue by abrkn** [lightningnetwork/lnd#1539]
### Background

```
ubuntu@ip-172-31-33-128:~$ lncli payinvoice lnbc1m1pd5vzpupp55mfmtlusuft3kkd7qk23aykkkkv0z8jlug5wnc0rnq32cmld39lqdz82pshjmt9de6zqen0wgsrzvpsxqcrqgrsd9ux2mrnypshggrnv96x7umgd9ejuurvv93k2tsxqzjccqp2nqam232xjk8786vzy35pq6jtkysnsezlalyyec267d4k2jlu7p8snk7752rrcndmjalc55zr6z5v7s3msj26cke3rjqaez05xxnea3gq0ugfhv
{
        "payment_error": "channel graph has insufficient capacity for the payment: need 0.001 BTC, have 0.001 BTC",
        "payment_preimage": "",
        "payment_route": null
}
```

### Your environment

```
ubuntu@ip-172-31-33-128:~$ lnd --version
lnd version 0.4.2-beta commit=a0b2fadea35fa4642daf4e9f56e6ecfe31d22ce1
ubuntu@ip-172-31-33-128:~$ lncli --version
lncli version 0.4.2 commit=a0b2fadea35fa4642daf4e9f56e6ecfe31d22ce1
ubuntu@ip-172-31-33-128:~$
```

Ubuntu 16.04

bitcoind

### Steps to reproduce

See above

### Expected behaviour

More informative error messages

### Actual behaviour

Confusing error message

**Comment by githorray** [comment]
I am seeing these as well on the latest master.

```
{
        "payment_error": "channel graph has insufficient capacity for the payment: need 0.002 BTC, have 0.002 BTC",
        "payment_preimage": "",
        "payment_route": null
}

real    0m0.111s
user    0m0.024s
sys     0m0.000s
```

**Comment by githorray** [comment]
I commented out the following code in pathfind.go and can now sling 500k sat payments across the network. #reckless #insanity

```
                // As a sanity check, we ensure that the incoming channel has
                // enough capacity to carry the required amount which
                // includes the fee dictated at each hop. Make the comparison
                // in msat to prevent rounding errors.
//              if currentHop.AmtToForward + fee > lnwire.NewMSatFromSatoshis(
//                      currentHop.Channel.Capacity) {
//
//                      err := fmt.Sprintf("channel graph has insufficient "+
//                              "capacity for the payment: need %v, have %v",
//                              currentHop.AmtToForward.ToSatoshis(),
//                              currentHop.Channel.Capacity)
//
//                      return nil, newErrf(ErrInsufficientCapacity, err)
//              }

```

**Comment by wpaulino** [comment]
This issue should be addressed by #1321.

**Comment by Roasbeef** [comment]
Fixed by #1321.
